### PR TITLE
CLI to build iron-iconsets based on a directory of SVGs

### PR DESCRIPTION
**Pull request by dlockhart**
@dbatiste: what do you think? This essentially takes a directory full of SVGs, and generates a WC HTML import that looks kind of like this:

```
<link rel="import" href="../polymer/polymer.html"/>
<link rel="import" href="../iron-iconset-svg/iron-iconset-svg.html"/>
<iron-iconset-svg name="vui-t3" size="30">
  <svg>
    <defs>
      <g id="bookmark">
        <path fill="none" d="stuff"/>
        <path d="stff"/>
      </g>
      <g id="chevron-down">
        <path d="stuff"/>
      </g>
    </defs>
  </svg>
</iron-iconset-svg>
```

This will let us have all our SVGs as files, and then auto-generate the WC imports. It iterates over the SVGs, grabs what's inside each `<svg>` element, REMOVES any "fill" attribute that isn't set to "none", and drops them all into `<g>` elements with their `id` set to the filename. It names the whole iconset `vui-{directoryName}`.

Finally, it uses the size of the first SVG to give the iconset a size... the idea being that all the icons in a directory are the same size.

### CODE REVIEW

File: iconset-builder.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			return deferred.reject(err);
+		}
+		var parser = new xml2js.Parser();
+		parser.parseString(xml, function(err, result) {
+			if (err) {
+				return deferred.reject(err);
+			}
+			if (iconsetObj['iron-iconset-svg'].$.size === '' && result.svg.$.viewBox) {
+				var values = result.svg.$.viewBox.split(' ');
+				iconsetObj['iron-iconset-svg'].$.size = values[values.length-1].toString();
```
Comment (dbatiste) [inline iconset-builder.js]: If I understand this correctly, we assume the image dimensions are square?  I suspect that's probably safe?  Also, I am guessing all the images we get from graphics folks can't have viewBox offsets, or we need to ensure those are corrected before putting into repo?

Comment (dlockhart) [inline iconset-builder.js]: Correct on both points. If either end up being not true, we can adjust... this was just the easiest way to get the value I needed and seems to work for all the icons I have so far.

**Comment by dbatiste** [comment]
Sounds good to me.  I'm not really a big fan of manually fixing the viewBox (if that's what we're planning to do).  I'm not sure what that process is, and I suspect it will be something that comes up over and over for others. It would be nice to handle that as part of this process, or if offsets are specified copy the viewBox forward to the `<g>` element since we know `iron-iconset-svg` will handle that.

File: iconset-builder.js (hunk 1 above)
Comment (dbatiste) [inline iconset-builder.js]: Cool. :)

**Comment by dlockhart** [comment]
Haha, I'll fix them for you if you want. It's super easy once you do it once, basically this:

```
<svg viewBox="-3 -4 24 24">
    <path />
</svg>
```

Becomes this:

```
<svg viewBox="0 0 24 24">
    <g transform="translate(3, 4)">
        <path />
    </g>
</svg>
```

Then you put it through SVGOmg and it optimizes everything away and removes the `<g>`.

**Comment by dbatiste** [comment]
Sweet!

**Pull request merged**